sanowret-crystal.lic

```diff
@@ -0,0 +1,72 @@
+=begin
+	Runs in the background, gazing or exhaling your sanowret crystal. Checks for concentration and current mindstate.
+
+	author: NBSL
+	game: dr
+  version: 0.1 (2017-06-12)
+
+  changelog:
+    0.1 (2017-06-12)
+     * Initial Release
+=end
+
+custom_require.call(%w[common drinfomon])
+
+
+class SanowretCrystal
+  def initialize
+    arg_definitions = [
+      [
+        { name: 'run', regex: /run/i, description: 'Single run of the script', optional: true }
+      ]
+    ]
+    @invalid_rooms = []
+    args = parse_args(arg_definitions)
+    check_crystal if args.run && !hiding?
+    passive unless args.run
+  end
+
+  def use_crystal
+    return if @invalid_rooms.include?(Room.current.id)
+    return if hiding?
+    if DRSkill.getxp('Arcana') <= 10
+      reponse = DRC.bput("gaze sanowret crystal", "A soft light blossoms in the very center of the crystal, and begins to fill your mind with the knowledge of \w+.", "However, nothing much else happens, as you lack the concentration to focus.", "This is not a good place for that.")
+    elsif DRSkill.getxp('Arcana') <= 25
+      response = DRC.bput("exhale sanowret crystal", "you come away from the experience with a further understanding of Arcana as the scintillating lights fade again.", "However, nothing much else happens, as you lack the concentration to focus.", "This is not a good place for that.", "Doing that would give away your hiding place.")
+    end
+
+    if response =~ "This is not a good place for that."
+      @invalid_rooms.push(Room.current.id) unless @invalid_rooms.include?(Room.current.id)
+      #move XMLData.room_exits.sample
+    end
+  end
+
+  def check_crystal
+    return if hiding?
+    if DRStats.concentration == 100 && DRSkill.getxp('Arcana') <= 25
+      case DRC.bput('tap my sanowret crystal', /^You tap a sanowret crystal inside your .*.$/,/^You tap a sanowret crystal that you are wearing.$/, /^You tap a sanowret crystal that you are holding.$/, "I could not find what you were referring to.")
+      when /You tap a sanowret crystal inside your/
+        fput("get my sanowret crystal")
+        use_crystal
+        fput("stow my sanowret crystal")
+      when /(^You tap a sanowret crystal that you are wearing.$|)/
```

Comment:
I removed that. It was left over from checking for 2 errors. As of right now I haven't pushed my latest changes, since I ran into issues with everything hanging when I'd try to ambush stun.